creusot-contracts/src/std/time.rs

```diff
@@ -107,6 +134,66 @@ extern_spec! {
                 #[ensures(rhs != 0u32 ==> result.deep_model() == Some(@self / @rhs))]
                 fn checked_div(self, rhs: u32) -> Option<Duration>;
             }
+
+            impl Instant {
+                #[ensures(@result >= 0)]
+                fn now() -> Instant;
+
+                #[ensures(@result >= 0)]
+                fn elapsed(&self) -> Duration;
+
+                #[ensures(@self > @earlier ==> @result > 0)]
+                #[ensures(@self <= @earlier ==> @result == 0)]
+                fn duration_since(&self, earlier: Instant) -> Duration;
+
+                #[ensures(@self >= @earlier ==> result != None)]
+                #[ensures(@self < @earlier ==> result == None)]
+                fn checked_duration_since(&self, earlier: Instant) -> Option<Duration>;
+
+                #[ensures(@self > @earlier ==> @result > 0)]
+                #[ensures(@self <= @earlier ==> @result == 0)]
+                fn saturating_duration_since(&self, earlier: Instant) -> Duration;
+
+                #[ensures(@duration == 0 ==> result.deep_model() == Some(@self))]
+                #[ensures(@duration > 0 && result != None ==> Some(@self) < result.deep_model())]
+                fn checked_add(&self, duration: Duration) -> Option<Instant>;
+
+                #[ensures(@duration == 0 ==> result.deep_model() == Some(@self))]
+                #[ensures(@duration > 0 && result != None ==> Some(@self) > result.deep_model())]
+                fn checked_sub(&self, duration: Duration) -> Option<Instant>;
```

Comment:
These specifications seems very weak. Is this intended?

Reply:
Instant is an opaque type with OS-specific behaviour which makes reasoning about its internal state close to impossible. But we can at least make some relative claims with these operations and that's what this specification captures.